Update fast/encoding/idn-security.html for new ICU versions

achristensen07 opened the pull request:
#### 2275c98a8611044a90faa270db6d72c4303f6161
<pre>
Update fast/encoding/idn-security.html for new ICU versions
<a href="https://bugs.webkit.org/show_bug.cgi?id=290866">https://bugs.webkit.org/show_bug.cgi?id=290866</a>
<a href="https://rdar.apple.com/148361142">rdar://148361142</a>

Reviewed by Anne van Kesteren.

UTS #46 version 16.0.0 included an update to the handling of several code points
when doing IDNA mapping, including U+115F, U+1160, U+3164, U+FFA0.  They are
mentioned in <a href="https://www.unicode.org/reports/tr46/tr46-31.html">https://www.unicode.org/reports/tr46/tr46-31.html</a> but not in
<a href="https://www.unicode.org/reports/tr46/tr46-33.html">https://www.unicode.org/reports/tr46/tr46-33.html</a> which was reflected in an ICU
update.

I updated the test to reflect the desired new behavior and mark it as failing on
old OSes that don&apos;t have the update.  I also remove the checks for &quot;older ICU&quot;
as of 2016, which is not supported or tested any more.

* LayoutTests/fast/encoding/idn-security-expected.txt:
* LayoutTests/fast/encoding/idn-security.html:
* LayoutTests/platform/mac-wk1/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/293093@main">https://commits.webkit.org/293093@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/29e3e689c8097e6aa4d3d9abe6deea2537ce4184

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/97799 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/17424 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/7641 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/102911 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/48328 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/17718 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/25877 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/74480 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/31657 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/100802 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/13414 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/88387 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/54835 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/13189 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/6307 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/47771 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/83183 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/6383 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/105291 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/24879 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/18123 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/83506 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/25251 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/84555 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/82942 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20934 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/27536 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/5213 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/18486 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/24840 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/24662 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/27976 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/26236 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk reviewed (approved):
I forgot about this test. Maybe I'll upstream this into WPT at some point as well so we don't have to handle it separately.

Thanks for tackling it!

achristensen07 committed:
Update fast/encoding/idn-security.html for new ICU versions
https://bugs.webkit.org/show_bug.cgi?id=290866
rdar://148361142

Reviewed by Anne van Kesteren.

UTS #46 version 16.0.0 included an update to the handling of several code points
when doing IDNA mapping, including U+115F, U+1160, U+3164, U+FFA0.  They are
mentioned in https://www.unicode.org/reports/tr46/tr46-31.html but not in
https://www.unicode.org/reports/tr46/tr46-33.html which was reflected in an ICU
update.

I updated the test to reflect the desired new behavior and mark it as failing on
old OSes that don't have the update.  I also remove the checks for "older ICU"
as of 2016, which is not supported or tested any more.

* LayoutTests/fast/encoding/idn-security-expected.txt:
* LayoutTests/fast/encoding/idn-security.html:
* LayoutTests/platform/mac-wk1/TestExpectations:

Canonical link: https://commits.webkit.org/293093@main
